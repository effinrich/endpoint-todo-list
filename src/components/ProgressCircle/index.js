import React from 'react'
import PropTypes from 'prop-types'
import Loader from 'react-loader'
import { useProgressBar } from 'react-aria'

import theme from 'theme'

import StyledLoader from './style'

const propTypes = {
  // expects #fff format
  color: PropTypes.string,
  scale: PropTypes.number,
  // relative or absolute
  position: PropTypes.string
}

const defaultProps = {
  color: theme.brandColor,
  scale: 0.25,
  position: 'relative',
  p: '9px'
}

const ProgressCircle = ({ color, scale, position, ...styledProps }, props) => {
  const { progressBarProps } = useProgressBar({
    isIndeterminate: true,
    'aria-label': 'Loading...'
  })

  return (
    <StyledLoader
      data-testid="progress-circle"
      {...progressBarProps}
      {...styledProps}
    >
      <Loader
        loaded={false}
        color={color}
        lines={10}
        length={20}
        width={10}
        radius={30}
        scale={scale}
        corners={1}
        direction={1}
        fps={20}
        zIndex={2e9}
        hwaccel={false}
        position={position}
      />
    </StyledLoader>
  )
}

ProgressCircle.propTypes = propTypes
ProgressCircle.defaultProps = defaultProps

export default ProgressCircle
