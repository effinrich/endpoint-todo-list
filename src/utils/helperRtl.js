import React from 'react'
import PropTypes from 'prop-types'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'

import store from 'store'

const Wrapper = ({ children }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  )
}

Wrapper.propTypes = {
  children: PropTypes.object
}

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
