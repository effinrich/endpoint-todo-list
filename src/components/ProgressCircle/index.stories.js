import React from 'react'

import theme from 'theme'

import ProgressCircle from '.'

export default {
  title: 'UI Library/ProgressCircle',
  component: ProgressCircle
}

const Template = (args) => <ProgressCircle {...args} />

export const Default = Template.bind({})
Default.args = { color: theme.brandColor }

export const Color = Template.bind({})
Color.args = {
  color: '#ff9900'
}

export const Scale = Template.bind({})
Scale.args = {
  scale: 0.756,
  color: theme.brandColor
}
