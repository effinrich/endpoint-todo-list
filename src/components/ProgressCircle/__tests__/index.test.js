import React from 'react'

import { render } from 'utils/testing/helperRtl'

import ProgressCircle from '..'

describe('ProgressCircle Component', () => {
  it('should render without crashing', () => {
    const { getByTestId } = render(<ProgressCircle />)
    expect(getByTestId('progress-circle')).toBeInTheDocument()
  })

  it('should render child element', () => {
    const { container } = render(<ProgressCircle />)
    expect(container.firstChild).toBeDefined()
  })
})
