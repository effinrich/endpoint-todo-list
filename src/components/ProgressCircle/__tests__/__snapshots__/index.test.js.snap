// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressCircle Component should render without crashing 1`] = `
<div>
  <div
    class="sc-bdfBwQ gQsXTp"
  >
    <div
      class="loader"
    >
      <div
        class="spinner"
        role="progressbar"
        style="position: relative; width: 0px; z-index: 2000000000; left: 50%; top: 50%;"
      >
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-0-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(0deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-1-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(36deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-2-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(72deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-3-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(108deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-4-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(144deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-5-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(180deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-6-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(216deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-7-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(252deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-8-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(288deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
        <div
          style="position: absolute; top: -1px; opacity: 0.25; webkit-animation: opacity-100-25-9-10 1s linear infinite;"
        >
          <div
            style="position: absolute; width: 7.5px; height: 2.5px; background: rgb(18, 146, 180); webkit-box-shadow: 0 0 1px rgba(0,0,0,.1); webkit-transform-origin: left; webkit-transform: rotate(324deg) translate(7.5px,0); webkit-border-radius: 1px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
